### PR TITLE
Internal Change: Code Cleanup

### PR DESCRIPTION
**Pull request by walkerselby**
Remove unnecessary checks
Remove unused local variables
Fix Typos
Add REPO todo
Add config getters
Merge nested if
Delete OtherMod, as API no longer in use

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ import kotlin.time.Duration.Companion.seconds
 class SlayerBossSpawnSoon {
 
     private val config get() = SkyHanniMod.feature.slayer.slayerBossWarning
-    private val pattern = " \\(?(?<progress>[0-9.,k]+)\\/(?<total>[0-9.,k]+)\\)?.*".toPattern()
+    private val pattern = " \\(?(?<progress>[0-9.,k]+)/(?<total>[0-9.,k]+)\\)?.*".toPattern()
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt:17]: I want to say these back slashes are still needed

Comment (walkerselby) [reply src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt:17]: > I want to say these back slashes are still needed

You might be right - I'll revert to be safe

File: src/main/java/at/hannibal2/skyhanni/features/inventory/ItemStars.kt [hunk 2]
```diff
@@ -23,7 +25,7 @@ class ItemStars {
 
         val stack = event.itemStack ?: return
         if (stack.stackSize != 1) return
```
Comment (Apo-th) [inline src/main/java/at/hannibal2/skyhanni/features/inventory/ItemStars.kt:26]: Why are they all separate ifs?

File: src/main/java/at/hannibal2/skyhanni/features/misc/MarkedPlayerManager.kt [hunk 3]
```diff
@@ -88,7 +90,7 @@ class MarkedPlayerManager {
     @SubscribeEvent
     fun onRenderMobColored(event: RenderMobColoredEvent) {
         if (!LorenzUtils.inSkyBlock) return
-        if (!SkyHanniMod.feature.markedPlayers.highlightInWorld) return
+        if (!config.highlightInWorld) return
```
Comment (Apo-th) [inline src/main/java/at/hannibal2/skyhanni/features/misc/MarkedPlayerManager.kt]: All separate ifs here too. Why?

File: src/main/java/at/hannibal2/skyhanni/features/inventory/ItemStars.kt (hunk 2 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/ItemStars.kt:26]: Better readability. But this was an older code style. I would try to merge them together now or extract into additional functions

File: src/main/java/at/hannibal2/skyhanni/features/misc/MarkedPlayerManager.kt (hunk 3 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/misc/MarkedPlayerManager.kt]: Here the usage of an isEnabled function is more consistent with the rest of the code base as well.

File: src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt:17]: Escaping the open bracket instead? Sound a bit regex behaviour manipulating

File: src/main/java/at/hannibal2/skyhanni/features/inventory/ItemStars.kt (hunk 2 above)
Comment (Apo-th) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/ItemStars.kt:26]: ah i see

File: src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt (hunk 1 above)
Comment (walkerselby) [reply src/main/java/at/hannibal2/skyhanni/features/slayer/SlayerBossSpawnSoon.kt:17]: @hannibal002 checked with nea and they said it is fine

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**